src/xmpp/session.c

```diff
@@ -196,20 +196,6 @@ session_connect_with_details(const char *const jid, const char *const passwd, co
 void
 session_autoping_fail(void)
 {
-    if (connection_get_status() == JABBER_CONNECTED) {
-        log_info("Closing connection");
-
-        char *account_name = session_get_account_name();
-        const char *fulljid = connection_get_fulljid();
-        plugins_on_disconnect(account_name, fulljid);
```

Comment:
I think we should still call this to make sure plugins can react on this?

Reply:
It is still called, I just removed redundant copy-paste:

session_autoping_fail() -> session_lost_connection() -> sv_ev_lost_connection() -> ev_disconnect_cleanup() -> session_disconnect().

And in the last function:
```C
    // if connected, send end stream and wait for response
    if (connection_get_status() == JABBER_CONNECTED) {
        log_info("Closing connection");

        char *account_name = session_get_account_name();
        const char *fulljid = connection_get_fulljid();
        plugins_on_disconnect(account_name, fulljid);
```

Reply:
Oh my bad! Thanks!